pallet/deposit/src/lib.rs

```diff
@@ -150,15 +150,9 @@ pub mod pallet {
 		pub fn lock(origin: OriginFor<T>, amount: Balance, months: u8) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			if amount < T::MinLockingAmount::get() {
-				Err(<Error<T>>::LockAtLeastSome)?;
-			}
-			if months == 0 {
-				Err(<Error<T>>::LockAtLeastOneMonth)?;
-			}
-			if months > 36 {
-				Err(<Error<T>>::LockAtMostThirtySixMonths)?;
-			}
+			ensure!(amount >= T::MinLockingAmount::get(), <Error<T>>::LockAtLeastSome);
```

Comment:
Just a personal idea.

I think the `if` is much easier to understand.

And the character count is nearly close. 😅

```rs
if amount < T::MinLockingAmount::get() { Err(<Error<T>>::LockAtLeastSome)?; }
ensure!(amount >= T::MinLockingAmount::get(), <Error<T>>::LockAtLeastSome);
```

Reply:
Because every time I saw the `ensure` I needed to do a bit-flipping in my mind. And to see what will cause this error.

Reply:
`ensure!` is such a very basic assert way in the substrate framework. It doesn't bring a lot of difficulty in understanding, but can be less `if`, so I think it's still acceptable.

Reply:
> Because every time I saw the ensure I had to do a bit-flipping in my mind.

Even with `if` branch  you still need to think about the condition and what triggers the error, I don't think it makes a difference. `ensure!` would make the code look a bit more compact, there are only 3 `if` branches here now, what if there are 5 or more, then multiple `if` branches in a row would look a bit ugly. That's my opinion, if more people agree with the if approach, I'll revert the changes here.

Reply:
I think that `if` is more readable here when reviewing code

Reply:
Okay. Let's me revert this.